src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -326,7 +326,7 @@ public boolean teleport(@NotNull Location location, @NotNull TeleportCause cause
 		boolean dismount = !flagSet.contains(TeleportFlag.EntityState.RETAIN_VEHICLE);
 		boolean ignorePassengers = flagSet.contains(TeleportFlag.EntityState.RETAIN_PASSENGERS);
 
-		if (flagSet.contains(TeleportFlag.EntityState.RETAIN_PASSENGERS) && this.hasPassengers() && location.getWorld() != this.getWorld())
+		if (flagSet.contains(TeleportFlag.EntityState.RETAIN_PASSENGERS) && this.hasPassengers() && location.getWorld().equals(this.getWorld()))
```

Comment:
The other way is also correct, both in MockBukkit and Paper

Reply:
`==`/`!=` compares identity while `Object#equals` compares content. Although the default implementation for `Object#equals` does just implement an identity check.

In addition, `a.equals(b)` will throw a null-pointer exception if `a` is null while `a == b` will never throw exceptions (although, I'm not sure of this actually. Perhaps it can throw a null pointer exception with automatic unboxing of a null?).

Null-pointer safe way of doing a content check: `Objects.equals(a, b)`